### ISSUE TITLE
Token errors all the time

### ISSUE THREAD

**Issue by ciscorucinski** [ganesh2shiv/yt-channel-list-android#1]
Pulling the code straight from GitHub and running the app does not display any videos. The `GetUsernameTask` will always have a `null` `result`  in `onTokenReceived()`. I am always presented with a Snackbar message of `Token error!`

Based on the comments in the code...

```
// The fetchToken() method handles Google-specific exceptions,
// so this indicates something went wrong at a higher level.
// TIP: Check for network connectivity before starting the AsyncTask. 
```

...this would appear to be a connectivity issue; however, I am connected online before I even open the app. Therefore, there must be something went wrong on a even higher level.

**Comment by ganesh2shiv** [comment]
Is google play services installed and updated on the device you are using?

**Comment by ciscorucinski** [comment]
I have `Version 8.4.89` installed and running on my phone

**Comment by ganesh2shiv** [comment]
Strange. Is youtube app installed on the device? Please share more details like model of the device and the logcat when you run the app so that I can test it from my end.

**Comment by ciscorucinski** [comment]
Yes, the YouTube App is installed on my device. I was able to get the Sample Google Sign In App, and the YouTube Sample app running on my phone perfectly fine.

I am working on my own YouTube app, but I don't understand why it is using the old, grey look and feel for the video UI, and not the new UI (wonder if these are connected somehow). I started to look at your code because I am trying to integrate Authentication with YouTube.

**Comment by ciscorucinski** [comment]
I have a `Samsung Galaxy Note 4` (`SM-N910L`) `Android 5.1.1`

[LogCat Output](http://pastebin.com/GvVMYKAD)

According to the output, it appears that the YouTube API Jar file is out of date, but I am using the newest version at 1.2.2

**Comment by ganesh2shiv** [comment]
Have you created the API keys (and enabled Youtube Data v3 API) from the Google developers console and put the generated API keys in the [Constants](https://github.com/ganesh2shiv/yt-channel-list-android/blob/master/app/src/main/java/app/vedicnerd/ytwua/util/Constants.java#L11) file?

If yes, have you tried debugging the code? I don't have access to Samsung Galaxy Note 4 so I am afraid I can't test it from my end. 

You should put a breakpoint at line 65 in [GetUsernameTask](https://github.com/ganesh2shiv/yt-channel-list-android/blob/master/app/src/main/java/app/vedicnerd/ytwua/async/GetUsernameTask.java#L65) class and then press F8 at every line while debugging. [This is the code block](https://github.com/ganesh2shiv/yt-channel-list-android/blob/master/app/src/main/java/app/vedicnerd/ytwua/async/GetUsernameTask.java#L65-L93) you should focus on because this is where your problem might be. 

Ideally the app should show you a confirmation dialog asking you to allow Youtube Channel list to manage/view your Youtube account for the first time only. Did it show any such dialog and did you allow it?

**Comment by ganesh2shiv** [comment]
I found the solution to your problem. 

It was because you were not creating the OAuth client ID for the project (which the Youtube data v3 API requires to access the user data). Just follow [these steps](https://github.com/ganesh2shiv/yt-channel-list-android#can-i-haz). 

I am closing this issue because I am pretty sure your problem would be resolved after following those steps. Feel free to create another issue if it doesn't work.
